File: backend/src/main/java/com/revnu/backend/features/auth/dto/AuthResponse.java
package com.revnu.backend.features.auth.dto;

import com.revnu.backend.features.auth.model.AccountStatus;
import com.revnu.backend.features.auth.model.RoleType;

public record AuthResponse(
        String message,
        String email,
        String fullname,
        RoleType role,
        AccountStatus status,
        String provider,
        boolean hasRestaurant,
        String accessToken
        ) {

}

File: mobile/app/src/main/java/com/revnu/mobile/features/auth/model/AuthResponse.kt
package com.revnu.mobile.features.auth.model

import com.google.gson.annotations.SerializedName
data class AuthResponse(
    @SerializedName("message")
    val message: String?,

    @SerializedName("email")
    val email: String,

    @SerializedName("fullname")
    val fullname: String?,

    @SerializedName("role")
    val role: String,

    @SerializedName("status")
    val status: String?,

    @SerializedName("provider")
    val provider: String?,

    @SerializedName("hasRestaurant")
    val hasRestaurant: Boolean,

    @SerializedName("accessToken")
    val accessToken: String,

    @SerializedName("refreshToken")
    val refreshToken: String
)